pkg/defaults/defaults.go

```diff
@@ -139,7 +139,18 @@ func setupCPUHotplug(clusterConfig *virtconfig.ClusterConfig, vmi *v1.VirtualMac
 	}
 
 	if vmi.Spec.Domain.CPU.MaxSockets == 0 {
+		// Define an implicit upper limit based on hotplug ratio (default: 128 * 4 = 512)
+		defaultMaxSocketsLimit := 128 * clusterConfig.GetMaxHotplugRatio()
```

Comment:
Where does 512 come from? Looking at the machine types provided upstream with QEMU 9.1.3 maxCpus ranges from 255 to 4096!

Reply:
AFAIK we only support q35, so let's just use its max?